### PR TITLE
[xray-exporter] Skip test case update if its already up to date in JIRA

### PR DESCRIPTION
**Pull request by uarlouski**

### CODE REVIEW

**Review by valfirst — CHANGES_REQUESTED** [review]

File: vividus-facade-jira/src/main/java/org/vividus/jira/databind/JiraEntityDeserializer.java [hunk 1]
```diff
@@ -36,13 +36,20 @@ public JiraEntity deserialize(JsonParser parser, DeserializationContext ctxt) th
         JsonNode jsonNode = parser.getCodec().readTree(parser);
 
         JiraEntity entity = new JiraEntity();
+        entity.setRaw(jsonNode.toString());
```
[the remark this answers is not in the document]
Comment (uarlouski) [reply vividus-facade-jira/src/main/java/org/vividus/jira/databind/JiraEntityDeserializer.java:39]: its a one redundant extra network call if bodies do not match

File: vividus-facade-jira/src/main/java/org/vividus/jira/JiraFacade.java [hunk 2]
```diff
@@ -106,14 +91,14 @@ public void addAttachments(String issueKey, List<Attachment> attachments)
...
             throws IOException, JiraConfigurationException
     {
-        return getJiraEntity(relativeUrl, entityKey, () -> jiraClientProvider.getByIssueKey(entityKey), entityType);
+        String response = requestEntity(relativeUrl, entityKey, () -> jiraClientProvider.getByIssueKey(entityKey));
+        return JiraEntityMapper.readValue(response, entityType);
     }
 
-    private <T> T getJiraEntity(String relativeUrl, String entityKey,
-            FailableSupplier<JiraClient, JiraConfigurationException> jiraClientSupplier, Class<T> entityType)
+    private String requestEntity(String relativeUrl, String entityKey,
```
Comment (valfirst) [inline vividus-facade-jira/src/main/java/org/vividus/jira/JiraFacade.java:98]:
```suggestion
    private String retrieveEntity(String relativeUrl, String entityKey,
```

File: vividus-facade-jira/src/main/java/org/vividus/jira/databind/JiraEntityDeserializer.java [hunk 3]
```diff
@@ -37,12 +37,18 @@ public JiraEntity deserialize(JsonParser parser, DeserializationContext ctxt) th
...
-                                             .readValueAs(new TypeReference<List<IssueLink>>() { });
+        JsonNode fields = jsonNode.path("fields");
+
+        List<IssueLink> issueLinks = fields.path("issuelinks")
+                                           .traverse(parser.getCodec())
+                                           .readValueAs(new TypeReference<List<IssueLink>>() { });
         entity.setIssueLinks(issueLinks);
+
+        String status = fields.path("status").path("name").asText();
+        entity.setStatus(status);
```
Comment (valfirst) [inline vividus-facade-jira/src/main/java/org/vividus/jira/databind/JiraEntityDeserializer.java:50]:
```suggestion
        entity.setStatus(fields.path("status").path("name").asText());
```

File: vividus-to-xray-exporter/build.gradle [hunk 4]
```diff
@@ -32,6 +32,8 @@ dependencies {
 
     implementation(group: 'org.zeroturnaround', name: 'zt-zip', version: '1.16')
 
+    implementation(group: 'net.javacrumbs.json-unit', name: 'json-unit', version: '3.2.2')
```
Comment (valfirst) [inline vividus-to-xray-exporter/build.gradle:35]: is `vividus-util` sill in use?

**Review by valfirst — CHANGES_REQUESTED** [review]

File: vividus-to-xray-exporter/src/main/java/org/vividus/xray/facade/XrayFacade.java [hunk 5]
```diff
@@ -196,14 +216,13 @@ public void updateTestSet(String testSetKey, List<String> testCaseKeys)
                 requestBody);
     }
 
-    private void checkIfIssueEditable(String issueKey)
+    private void checkIfIssueEditable(JiraEntity jiraEntity)
             throws IOException, NonEditableIssueStatusException, JiraConfigurationException
```
Comment (valfirst) [inline vividus-to-xray-exporter/src/main/java/org/vividus/xray/facade/XrayFacade.java:220]:
```suggestion
            throws NonEditableIssueStatusException
```

**Pull request merged**